backend/app/services/adapters/bot_kinds.py

```
@@ -554,40 +551,31 @@ def update_with_user(
-                        PublicShell.name == new_agent_name,
-                        PublicShell.namespace == "default",
-                    )
-                    .first()
-                )
+        if "shell_name" in update_data:
+            # Update Bot's shellRef to point directly to the user-selected Shell
+            new_shell_name = update_data["shell_name"]
+            try:
+                shell_info = get_shell_info_by_name(db, new_shell_name, user_id)
+            except ValueError as e:
+                raise HTTPException(status_code=400, detail=str(e))
```

Ruff (0.14.6)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)